esiclient/v1/node_network.py

```diff
@@ -149,17 +165,22 @@ def take_action(self, parsed_args):
             raise exceptions.CommandError(
                 "ERROR: Node {0} has no free ports".format(node.name))
 
-        print("Attaching network {1} to node {0}".format(
-            node.name, network.name))
-
-        port = neutron_client.create_port(name=node.name,
-                                          network_id=network.id)
-        ironic_client.node.vif_attach(node_uuid, port.id)
-        updated_port = neutron_client.get_port(port.id)
+        if port:
+            print("Attaching node {0} to port {1}".format(
+                node.name, port.name))
+            ironic_client.node.vif_attach(node_uuid, port.id)
+            network = neutron_client.get_network(port.network_id)
+        else:
+            print("Attaching network {1} to node {0}".format(
+                node.name, network.name))
+            port = neutron_client.create_port(name=node.name,
+                                              network_id=network.id)
+            ironic_client.node.vif_attach(node_uuid, port.id)
+            port = neutron_client.get_port(port.id)
```

Comment:
probably a dumb question but it's not clear to me why are we either attaching a port to a node or a network to a node.

Reply:
Solely for ease of use - it's more intuitive (I think) to be able to attach a network to a node and have the underlying code create a port for you. However things like trunk ports require being able to attach a network to an existing port. I could see the network-to-node option being dropped though!

Reply:
> it's more intuitive (I think) to be able to attach a network to a node and have the underlying code create a port for you.

That makes sense.

> However things like trunk ports require being able to attach a network to an existing port. 

I think at some point we should have add a `docs` directory in this repo to document how to use the CLI.

Reply:
100% agree! I haven't done it yet because it feels like the CLI could still be in considerable flux. But I'll add a trello task to make sure it gets done.